pyVHDLParser/Blocks/Reference/Context.py

```diff
@@ -27,139 +27,354 @@
 # limitations under the License.                                                                                       #
 # ==================================================================================================================== #
 #
+from typing                           import Type, Dict
+
 from pyTooling.Decorators             import export
 
-from pyVHDLParser.Token               import CommentToken, WhitespaceToken, LinebreakToken, MultiLineCommentToken, IndentationToken, SingleLineCommentToken, ExtendedIdentifier
-from pyVHDLParser.Token.Keywords      import WordToken, BoundaryToken, IdentifierToken, IsKeyword, UseKeyword, EndKeyword, ContextKeyword, LibraryKeyword
-from pyVHDLParser.Blocks              import Block, CommentBlock, BlockParserException, TokenToBlockParser
-from pyVHDLParser.Blocks.Whitespace       import LinebreakBlock, IndentationBlock, WhitespaceBlock
-from pyVHDLParser.Blocks.Region      import EndBlock as EndBlockBase
+from pyVHDLParser.Token               import Token, CommentToken, WhitespaceToken, LinebreakToken, IndentationToken, ExtendedIdentifier, CharacterToken, ValuedToken
+from pyVHDLParser.Token.Keywords      import WordToken, IdentifierToken, IsKeyword, UseKeyword, EndKeyword, ContextKeyword, LibraryKeyword, EndToken, DelimiterToken, KeywordToken
+from pyVHDLParser.Blocks              import Block, BlockParserException, TokenToBlockParser, SkipableBlock, BLOCK_PARSER_STATE
+from pyVHDLParser.Blocks.Region       import EndBlock as EndBlockBase
+
+
+def _isNonCodeToken(token: Token) -> bool:
+	return isinstance(token, (LinebreakToken, WhitespaceToken, IndentationToken, CommentToken))
+
+
 
 
 @export
-class NameBlock(Block):
-	KEYWORDS = None
+class DelimiterBlock(SkipableBlock):
+	pass
 
-	@classmethod
-	def __cls_init__(cls):
-		from pyVHDLParser.Blocks.Reference     import Use, Library
 
-		cls.KEYWORDS = {
-			# Keyword       Transition
-			UseKeyword:     Use.StartBlock.stateUseKeyword,
-			LibraryKeyword: Library.StartBlock.stateLibraryKeyword
-		}
+@export
+class StartBlock(Block):
+	"""
+	This block is used to disambiguate between context declarations and context references.
+
+	The states in this class only check whether the token stream represents a declaration or a reference.
+	They neither change :class:`Token`s nor do they create new :class:`Block`s.
+	After a decision can be made, the unhandled tokens (everything between :attr:`parserState.TokenMarker` and the current token)
+	are parsed a second time by either :attr:`ReferenceStartBlock.fromTokenMarker` or :attr:`DeclarationStartBlock.fromTokenMarker`.
+	"""
 	@classmethod
 	def stateContextKeyword(cls, parserState: TokenToBlockParser):
+		cls.stateWhitespace1(parserState)

```

Comment:
It seems like you named all your states after the token which was expected in the previous state, which I find a little counter-intuitive. 

For the sake of a consistent interface I kept that naming scheme for the first state, and named all subsequent states after the token they are expecting right now.